### ISSUE TITLE
custom check with gradle

### ISSUE THREAD

**Issue by kdh6429** [checkstyle/checkstyle#12988]
I have read check documentation: https://checkstyle.org/writingchecks.html
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ cat CustomCheck.java
# package com.example.checks;

# import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
# import com.puppycrawl.tools.checkstyle.api.DetailAST;
# public class CustomCheck extends AbstractCheck {
#  ...
# }

/var/tmp $ cat config/checkstyle.xml
# <module name="com.example.checks.CustomCheck">
# ...
```
---
<img width="414" alt="image" src="https://user-images.githubusercontent.com/6263966/232224945-3139b1ac-0f0c-4fc8-bace-ac69efc1c48a.png">

I'm trying to do it with gradle, not the command with` java -classpath .. `When I ran gradle checkstyleMain , I got the following error.  Can't I do it like this?

> Unable to instantiate 'com.example.checks.CustomCheck' class, it is also not possible to instantiate it as null. Please recheck that class name is specified as canonical name or read how to configure short name usage https://checkstyle.org/config.html#Packages. Please also recheck that provided ClassLoader to Checker is configured correctly.

---

ATTENTION: FAILURE TO FOLLOW THE ABOVE TEMPLATE WILL RESULT IN THE ISSUE BEING CLOSED.

**Comment by romani** [comment]
you need fix problem of class path during run of some gradle task. as your Check need to be compiled and all dependencies also added to class path to let it work. It is not easy, you need to open issue in Gradle community or stackoverflow to figureout this hack.

Checkstyle project do similar by means of ant:
https://github.com/checkstyle/checkstyle/blob/9ee3d5445268392c7f17a1a5a7f034cdc44ab5b0/config/ant-phase-verify.xml#L14-L16
(very special operation on classpath)

if you move your check out of application it will easier to do this, here is example https://github.com/sevntu-checkstyle/checkstyle-samples/blob/1615ad3ff3aabfe6b7c5e16ce85e41763f4d4c25/gradle-project/checkstyle.xml#L10 you can find custom Check at https://github.com/sevntu-checkstyle/sevntu.checkstyle/tree/master/sevntu-checks
